custom/icds_reports/utils/aggregation.py

```diff
@@ -500,12 +500,15 @@ def _old_ucr_tablename(self):
         config, _ = get_datasource_config(doc_id, self.domain)
         return get_table_name(self.domain, config.table_id)
 
-    def _data_from_ucr_query(self, column=None):
+    def _data_from_ucr_query(self, column=None, uses_zero_as_null=False):
         current_month_start = month_formatter(self.month)
         next_month_start = month_formatter(self.month + relativedelta(months=1))
         if column is not None:
             select_line = ", LAST_VALUE({column}) OVER w AS {column}".format(column=column)
-            where_line = " AND {column} IS NOT NULL".format(column=column)
+            if uses_zero_as_null:
```

Comment:
Would this logic work for cases besides the zscore, like height and weight? In those cases, are they just never recorded as null? If so, are you setting this boolean just to be more explicit?

Reply:
This would not work for height and weight because if they are not recorded it's NULL. The general workflow I've been following is to have NULL mean not recorded. The reason  that NULL doesn't work for zscore is because there's an explicit [switch statement in the data source](https://github.com/dimagi/commcare-hq/blob/38c305c0290e957754efc71634a10b9a2171a62f/custom/icds_reports/ucr/data_sources/dashboard/dashboard_growth_monitoring.json#L232-L249) that coerces string values into an integer.

Reply:
cool, thanks